src/main/kotlin/domain/racingcar/component/Race.kt

```diff
@@ -0,0 +1,27 @@
+package domain.racingcar.component
+
+import domain.racingcar.RacingCarContainer
+import domain.racingcar.dto.RaceHistoryDto
+import domain.racingcar.dto.StepHistoryDto
+import domain.racingcar.strategy.RaceStrategy
+import domain.racingcar.strategy.RandomStrategy
+
+class Race(
+    private val racingcarQuantity: Int,
+    private val totalStep: Int,
+    private val raceStrategy: RaceStrategy
+) {
+    private val raceHistory = mutableListOf<StepHistoryDto>()
+
+    fun start(): RaceHistoryDto {
+
+        val racingCarContainer = RacingCarContainer(racingcarQuantity, raceStrategy)
+
+        for (i in 1..totalStep) {
```

Comment:
```suggestion
        (0 until totalStep).map { racingCarContainer.stepProgress() }
```
한 줄로 작성할수도 있을 것 같네요. 더 합치자면

```kotlin
        (0 until totalStep).map { racingCarContainer.stepProgress() }.let(::RaceHistoryDto)
```

이렇게도 표현 가능하겠네요